lib/streamlit/elements/time_widgets.py

```diff
@@ -33,12 +34,138 @@
 from .form import current_form_id
 from .utils import check_callback_rules, check_session_state_rules
 
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+
+TimeValue: TypeAlias = Union[time, datetime, None]
+_DateValue: TypeAlias = Union[date, datetime, None]
+DateValue: TypeAlias = Union[_DateValue, Sequence[_DateValue]]
```

Comment:
The naming of `_DateValue` and `DateValue` could be improved here to differentiate a bit better. Maybe something like `SingleDateValue` + `DateValue` or `DateValue` + `DateWidgetValue`.

Reply:
Agreed. I think `_DataValue` started out as just a useful construct for defining `DateValue`, but since it started showing up in signatures in its own right, it makes sense to give it a proper name. Went with the `SingleDateValue` approach.